corehq/apps/locations/tests/test_location_fixtures.py

```diff
@@ -258,6 +263,105 @@ def test_index_location_fixtures(self):
+        ('Massachusetts', [
+            ('Middlesex', [
+                ('Cambridge', []),
+                ('Somerville', []),
+            ]),
+            ('Suffolk', [
+                ('Boston', []),
+                ('Revere', []),
+            ])
+        ]),
+    ]
+
+    @classmethod
+    def setUpClass(cls):
+        super(LocationFixturesDataTest, cls).setUpClass()
+        cls.user = create_restore_user(cls.domain, 'user', '123')
+        # cls.addCleanup(cls.user._couch_user.delete)
+        cls.loc_fields = CustomDataFieldsDefinition.get_or_create(cls.domain, LocationFieldsView.field_type)
+        cls.loc_fields.fields = [
+            CustomDataField(slug='baseball_team'),
+            CustomDataField(slug='favorite_passtime'),
+        ]
+        cls.loc_fields.save()
+        cls.field_slugs = {f.slug for f in cls.loc_fields.fields}
+
+    def setUp(self):
+        # this works around the fact that get_locations_to_sync is memoized on OTARestoreUser
+        self.user = self.user._couch_user.to_ota_restore_user()
+
+    @classmethod
+    def tearDownClass(cls):
+        cls.loc_fields.delete()
+        cls.user._couch_user.delete()
+        super(LocationFixturesDataTest, cls).tearDownClass()
+
+    def test_utility_method(self):
+        self.assertEqual(self.field_slugs, _get_location_data_fields(self.domain))
+
+    def test_metadata_added_to_all_nodes(self):
+        mass = self.locations['Massachusetts']
+        self.user._couch_user.set_location(mass)
+        fixture = flat_location_fixture_generator(self.user, V2)[1]  # first node is index
+        location_nodes = fixture.findall('locations/location')
+        self.assertEqual(7, len(location_nodes))
+        for location_node in location_nodes:
+            location_data_nodes = [child for child in location_node.find('location_data')]
+            self.assertEqual(2, len(location_data_nodes))
+            tags = {n.tag for n in location_data_nodes}
+            self.assertEqual(tags, self.field_slugs)
+
+    def test_additional_metadata_included(self):
+        mass = self.locations['Massachusetts']
+        mass.metadata = {'driver_friendliness': 'poor'}
+        mass.save()
+
+        def _clear_metadata():
+            mass.metadata = {}
+            mass.save()
+
+        self.addCleanup(_clear_metadata)
+        self.user._couch_user.set_location(mass)
+        fixture = flat_location_fixture_generator(self.user, V2)[1]  # first node is index
+        mass_data = [
+            field for field in fixture.find('locations/location[@id="{}"]/location_data'.format(mass.location_id))
+        ]
+        self.assertEqual(3, len(mass_data))
+        self.assertEqual(self.field_slugs | set(['driver_friendliness']), set([f.tag for f in mass_data]))
+        self.assertEqual(
+            'poor',
+            fixture.find(
+                'locations/location[@id="{}"]/location_data/driver_friendliness'.format(mass.location_id)
+            ).text
+        )
```

Comment:
Yeah, I think we should skip that data.  I'd be concerned that these apps might break anyways if the fixture is heterogeneous because of non-schema data.